ale_linters/yaml/actionlint.vim

```diff
@@ -21,15 +21,27 @@ endfunction
 function! ale_linters#yaml#actionlint#Handle(buffer, lines) abort
     " Matches patterns line the following:
     ".github/workflows/main.yml:19:0: could not parse as YAML: yaml: line 19: mapping values are not allowed in this context [yaml-syntax]
-    let l:pattern = '\v^.*:(\d+):(\d+): (.+) \[(.+)\]$'
+    let l:pattern = '\v^.{-}:(\d+):(\d+): (.+) \[(.+)\]$'
     let l:output = []
 
+
     for l:match in ale#util#GetMatches(a:lines, l:pattern)
+        let l:code = l:match[4]
+        let l:text = l:match[3]
+
+        " Handle sub-linter errors like the following:
```

Comment:
I extracted the `code` and `text` in their own variables, to overwrite them when in the `shellcheck` special case. If so, I would parse the text to only focus on the shellcheck error message and set the code to `shellcheck SCXXXX` (where XXXX is the shellcheck error code, as per their documentation)